src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -517,6 +518,7 @@ internal UnboundLambdaState WithCaching(bool includeCache)
         public UnboundLambda UnboundLambda => _unboundLambda;
 
         public abstract MessageID MessageID { get; }
+        public abstract bool HasParameterNames();
```

Comment:
>HasParameterNames() [](http://example.com/codeflow?start=29&length=19)

Is this distinct from `HasNames` or `HasSignature`?

Reply:
Looks like we have a triplicate. I'll remove two